pkg/component/observability/opentelemetry/collector/collector.go

```diff
@@ -0,0 +1,162 @@
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/component"
+	valiconstants "github.com/gardener/gardener/pkg/component/observability/logging/vali/constants"
+	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+)
+
+const (
+	managedResourceName            = "opentelemetry-collector"
+	otelCollectorConfigName        = "opentelemetry-collector-config"
+	timeoutWaitForManagedResources = 2 * time.Minute
+)
+
+// Values is the values for otel-collector configurations
+type Values struct {
+	// Image is the collector image.
+	Image string
+}
+
+type otelCollector struct {
+	client       client.Client
+	namespace    string
+	values       Values
+	lokiEndpoint string
+}
+
+// New creates a new instance of otel-collector deployer.
+func New(
+	client client.Client,
+	namespace string,
+	values Values,
+	lokiEndpoint string,
+) component.DeployWaiter {
+	return &otelCollector{
+		client:       client,
+		namespace:    namespace,
+		values:       values,
+		lokiEndpoint: lokiEndpoint,
+	}
+}
+
+func (o *otelCollector) Deploy(ctx context.Context) error {
+	registry := managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
+
+	serializedResources, err := registry.AddAllAndSerialize(o.openTelemetryCollector(o.namespace, o.lokiEndpoint))
+	if err != nil {
+		return err
+	}
+
+	return managedresources.CreateForSeedWithLabels(ctx, o.client, o.namespace, managedResourceName, false, map[string]string{v1beta1constants.LabelCareConditionType: v1beta1constants.ObservabilityComponentsHealthy}, serializedResources)
+}
+
+func (o *otelCollector) Destroy(ctx context.Context) error {
+	return managedresources.DeleteForSeed(ctx, o.client, o.namespace, managedResourceName)
+}
+
+func (o *otelCollector) Wait(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, timeoutWaitForManagedResources)
+	defer cancel()
+
+	return managedresources.WaitUntilHealthy(timeoutCtx, o.client, o.namespace, managedResourceName)
+}
+
+func (o *otelCollector) WaitCleanup(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, timeoutWaitForManagedResources)
+	defer cancel()
+
+	return managedresources.WaitUntilDeleted(timeoutCtx, o.client, o.namespace, managedResourceName)
+}
+
+func (o *otelCollector) openTelemetryCollector(namespace, lokiEndpoint string) *otelv1beta1.OpenTelemetryCollector {
+	return &otelv1beta1.OpenTelemetryCollector{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "opentelemetry-collector",
+			Namespace: namespace,
+			Labels:    getLabels(),
+		},
+		Spec: otelv1beta1.OpenTelemetryCollectorSpec{
+			Mode:            "deployment",
+			UpgradeStrategy: "none",
+			OpenTelemetryCommonFields: otelv1beta1.OpenTelemetryCommonFields{
+				Image: o.values.Image,
+			},
```

Comment:
```suggestion
			OpenTelemetryCommonFields: otelv1beta1.OpenTelemetryCommonFields{
				Image: o.values.Image,
				SecurityContext: &corev1.SecurityContext{
					AllowPrivilegeEscalation: ptr.To(false),
				},
			},
```

Can we also forbid privilege escalation for the otel collector?

Reply:
Absolutely. Thank you for catching this before the PR was accepted.